tests/dbt/test_integration.py

```diff
@@ -0,0 +1,347 @@
+import shutil
+import typing as t
+from enum import Enum
+from functools import partial
+from pathlib import Path
+
+import numpy as np
+import pandas as pd
+import pytest
+from dbt.cli.main import dbtRunner
+from freezegun import freeze_time
+from sqlglot import exp
+
+from sqlmesh import Context
+from sqlmesh.core.config.connection import DuckDBConnectionConfig
+from sqlmesh.core.engine_adapter import DuckDBEngineAdapter
+from sqlmesh.utils.date import now
+from sqlmesh.utils.yaml import YAML
+
+pytestmark = [pytest.mark.dbt, pytest.mark.slow]
+
+
+class TestType(str, Enum):
+    DBT_RUNTIME = "dbt_runtime"
+    DBT_ADAPTER = "dbt_adapter"
+    SQLMESH = "sqlmesh"
+
+    @property
+    def is_sqlmesh(self) -> bool:
+        return self == TestType.SQLMESH
+
+    @property
+    def is_dbt_runtime(self) -> bool:
+        return self == TestType.DBT_RUNTIME
+
+    @property
+    def is_dbt_adapter(self) -> bool:
+        return self == TestType.DBT_ADAPTER
+
+    @property
+    def is_dbt_project(self):
+        return self.is_dbt_adapter or self.is_dbt_runtime
+
+    @property
+    def is_sqlmesh_runtime(self) -> bool:
+        return self.is_sqlmesh or self.is_dbt_adapter
+
+
+@pytest.fixture(params=list(TestType))
+def test_type(request):
+    return request.param
+
+
+@pytest.fixture
+def create_scd_type_2_dbt_project(tmp_path):
```

Comment:
Could we just add an scd type 2 model in the the dbt sushi project? I'd rather have these big model types in those examples because they're run all over the place.

In general, I'm afraid of the maintainability of these custom integration tests, which is why I lean towards adding these directly to the test projects.

Reply:
The test dbt sushi project does have an scd type 2 model. It actually has 3 of them (to verify what does or does not work). 

Lets discuss the maintainability concern.